src/Val.h

```diff
@@ -1673,26 +1672,17 @@ class VectorVal final : public Val, public notifier::detail::Modifiable
 	 * @param index  The position in the vector of the element to return.
 	 * @return  The element's underlying value.
 	 */
-	zeek_int_t IntAt(unsigned int index) const { return (*vector_val)[index]->int_val; }
-	zeek_uint_t CountAt(unsigned int index) const { return (*vector_val)[index]->uint_val; }
-	double DoubleAt(unsigned int index) const { return (*vector_val)[index]->double_val; }
-	const RecordVal* RecordValAt(unsigned int index) const
-		{
-		return (*vector_val)[index]->record_val;
-		}
-	bool BoolAt(unsigned int index) const
-		{
-		return static_cast<bool>((*vector_val)[index]->uint_val);
-		}
-	const StringVal* StringValAt(unsigned int index) const
-		{
-		return (*vector_val)[index]->string_val;
-		}
+	zeek_int_t IntAt(unsigned int index) const { return vector_val[index]->int_val; }
+	zeek_uint_t CountAt(unsigned int index) const { return vector_val[index]->uint_val; }
+	double DoubleAt(unsigned int index) const { return vector_val[index]->double_val; }
+	const RecordVal* RecordValAt(unsigned int index) const { return vector_val[index]->record_val; }
+	bool BoolAt(unsigned int index) const { return static_cast<bool>(vector_val[index]->uint_val); }
+	const StringVal* StringValAt(unsigned int index) const { return vector_val[index]->string_val; }
 	const String* StringAt(unsigned int index) const { return StringValAt(index)->AsString(); }
 
 	// Only intended for low-level access by internal or compiled code.
-	const auto& RawVec() const { return vector_val; }
-	auto& RawVec() { return vector_val; }
+	const std::vector<std::optional<ZVal>>& RawVec() const { return vector_val; }
+	std::vector<std::optional<ZVal>>& RawVec() { return vector_val; }
```

Comment:
API break here intentionally: This previously returned `std::vector<std::optional<ZVal>>*&` (a reference to a pointer to a vector). Mainly taking the comment above as justification for breaking (and most users primarily only skipping the de-referencing of the pointer).

```
	// Only intended for low-level access by internal or compiled code.
```